app/src/main/java/com/bogdan/codeforceswatcher/adapter/ContestAdapter.kt

```diff
@@ -24,6 +27,18 @@ class ContestAdapter(private var items: List<Contest>, private val ctx: Context)
         val contest = items[position]
         holder.tvContestName.text = contest.name
         holder.tvContestTime.text = getDateTime(contest.time)
+        holder.ivAddToCalendar.setOnClickListener { addContestToCalendar(items[position]) }
+    }
+
+    private fun addContestToCalendar(contest: Contest) {
+        val contestName = contest.name.replace(' ', '+')
```

Comment:
I think, what you've tried to do here is to "encode url". If so, google it and you will find a standard way of doing it.